docs/SandboxExtention.md

```diff
@@ -0,0 +1,71 @@
+# OTP Sandbox Extensions
+
+- The sandbox is a place to test and implement new "experimental" features.  
+- This should not be used for bug fixes and smaller changes.
+- Consider forking if the feature is valuable to one deployment only.
+
+
+## Available extensions
+Here is a list of features implemented as OTP Sandbox Extensions. TODO - Disclaimer on usage.
+
+- [Examples](../src/ext/java/org/opentripplanner/ext/examples/README.md) - Sandbox examples on how 
+  to implement extensions.
+
+
+## Terminology
+
+**Main/core**   -- All OTP code and additional files, NOT part of the sandbox. 
+(`docs`, `src/main`, `src/test` and so on)
+
+**Extensions** -- All features implemented in the OTP Sandbox, provided with no guarantees. 
+(`src/ext`, `src/ext-test`)  
+
+## Sandbox Goals
+
+- Reduce work for PR approval
+- Allow experimental code to evolve (in a Sandbox)
+- Encourage refactoring and creation of extension points in the main code.
+- Increase visibility and cooperation of development of new features.
+- Feature toggle 
+  - Default is off
+  - In the config file(s) there should be a list of enabled extensions. To enable an extension add 
+  the name of the extension to the list.
+
+
+## Contract
+- Give your feature a name: `<extension name>`
+- A new feature is isolated from the rest of the code by putting it in the directory `src/ext`. 
+  Java code should have package prefix `org.opentripplanner.ext.<extension name>`. Unit tests 
+  should be added in the test directory: `src/ext-test`
+- To integrate the new feature into OTP you may have to create new extension points in the main 
+  code. 
+- Create a readme file (`src/ext/java/org/opentripplanner/ext/<extension name>/<Extension Name>.md` 
+  package including:
+    - Contact info 
+    - Change log
+    - Documentation of the feature (optional)
+- List your extension in the [Available extensions](#Available extensions) section.
+- Use feature toggling to to enable a feature at runtime. The feature must be disabled by default. 
+  A feature is toggled _on_ using the config files.
+- Only code modifying the main code(`src/main`, not `src/ext`) is reviewed. The current coding 
+  standard apply to the extension code as well - but the code is not necessarily reviewed.
+- There are no grantees - the authors of an extension can change its API any time they want. 
+- Anyone can request the feature to be merged into the main code. An approval from the PLC and a 
+  new review is then required. The reviewers may request any changes, including API changes. 
+- If an extension is taken into core/main OTP the API may change, no BACKWARD compatibility is 
+  guaranteed. 
+- The feature submitters is responsible for maintaining and testing the extension code, but do not
+  need to provide any guarantees or support. 
```

Comment:
This applies while it is an extension. Merging to non extension area implies that the author will in fact provide support and maintenance.